### PR TITLE
Updated fetcher in order not to do useless jobs

### PR DESCRIPTION
**Pull request by jeongkyun-oh**
## Proposed changes

- This PR is derived from ethereum/go-ethereum#21263
- The fetcher did a lot of useless work, which is simplified.

## Types of changes

Please put an x in the boxes related to your change.

- [ ] Bugfix
- [x] New feature or enhancement
- [ ] Others

## Checklist

*Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.*

- [x] I have read the [CONTRIBUTING GUIDELINES](https://github.com/klaytn/klaytn/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla-assistant.io/klaytn/klaytn)
- [x] Lint and unit tests pass locally with my changes (`$ make test`)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: datasync/fetcher/fetcher.go [hunk 1]
```diff
@@ -551,33 +551,44 @@ func (f *Fetcher) loop() {
...
+						if txnHash != announce.header.TxHash {
+							continue
+						}
+
+						// Mark the body matched, reassemble if still unknown
+						matched = true
+						if f.getBlock(hash) == nil {
+							block := types.NewBlockWithHeader(announce.header).WithBody(task.transactions[i])
+							block.ReceivedAt = task.time
+
```
Comment (ehnuje) [inline datasync/fetcher/fetcher.go:581]:
```suggestion
```
Unnecessary white space (also different from geth)?

**Review by ehnuje — APPROVED** [review]

File: datasync/fetcher/fetcher.go (hunk 1 above)
Comment (jeongkyun-oh) [reply datasync/fetcher/fetcher.go]: Committed :)

**Review by KimKyungup — APPROVED** [review]
This is not whole change of https://github.com/ethereum/go-ethereum/pull/21263.
Will you apply all change of it?

If not, please note that you applied some part of the PR on the excel sheet. ^^

**Comment by jeongkyun-oh** [comment]
@KimKyungup Yes, I am planning to apply some others. Those PR is too big, so I dividing several ones. It is taking time to understand what PR contains. Anyway, I will update the spread sheet as well.

**Review by ehnuje — APPROVED** [review]

**Review by yoomee1313 — APPROVED** [review]

**Pull request merged**